### DISCUSSION TITLE
Add TOC via shortcode

### DISCUSSION THREAD

**Discussion by simplerethink** [11ty/eleventy#2588] [Q&A]
I'm trying to add a table of contents to markdown pages. I know there are several 11ty TOC plugins for this, but it seems they need to be placed in template files. I'm trying to specify a TOC within markdown content using a shortcode so I can specify the location within each article. Is this possible?

**Comment by pdehaan** [comment]
There are a lot of eleventy and/or markdown-it plugins for this. Here are two random ones I found that seem to support arbitrary placed table of contents blocks:

- https://www.npmjs.com/package/markdown-it-table-of-contents (via `[[toc]]`)
- https://www.npmjs.com/package/eleventy-plugin-toc (via `{{ content | toc }}`)
